### PR TITLE
Change X.509 test case to not rely on asn1parse limitation

### PR DESCRIPTION
**Pull request by gilles-peskine-arm**
* Tweak test data for one test case to not rely on `mbedtls_asn1_get_int` lacking support for leading zeros.
* Tweak test data for one test case to not rely on `mbedtls_asn1_get_bitstring_null` rejecting bitstrings shorter than two octets.

Needed for bug fixes in `mbedtls_asn1_get_int` and `mbedtls_asn1_get_bitstring_null` in https://github.com/ARMmbed/mbed-crypto/pull/75. The updated test data works both with the current ASN.1 implementation and with the fixes from https://github.com/ARMmbed/mbed-crypto/pull/75.

One new test (empty Signature) is commented out because it needs the fix from https://github.com/ARMmbed/mbed-crypto/pull/75. I included it in this PR because it's natural to have it given the other test data, but currently `mbedtls_x509_crt_parse` would return an erroneous error code due to the `mbedtls_asn1_get_bitstring_null` limitation.

### CODE REVIEW

**Comment by gilles-peskine-arm** [comment]
@k-stachowiak I'm sorry, I don't understand your comment. This PR is not supposed to replace a negative test by a positive test. Each test is still supposed to exercise a case where a particular call to an `mbedtls_asn1_xxx` function returns an error. What changes is that those two tests relied on a bug in that `mbedtls_asn1_xxx` function. I've changed the test data to trigger the same error legitimately rather than due to a bug.

**Comment by gilles-peskine-arm** [comment]
@Patater Why "needs: preceding PR"? This PR needs to go in before https://github.com/ARMmbed/mbed-crypto/pull/75.

File: tests/suites/test_suite_x509parse.data [hunk 1]
```diff
@@ -1825,13 +1826,36 @@ X509 CRT ASN1 (inv Signature, length out of bounds)
...
+# signature = bit string with invalid encoding (number of unused bits too large)
+x509parse_crt:"3081bc3081a7a0030201028204deadbeef300d06092a864886f70d01010b0500300c310a30080600130454657374301c170c303930313031303030303030170c303931323331323335393539300c310a30080600130454657374302a300d06092A864886F70D010101050003190030160210ffffffffffffffffffffffffffffffff0202ffffa100a200a321301f301d0603551d11041630148208666f6f2e7465737482086261722e74657374300d06092a864886f70d01010b0500030108":"":MBEDTLS_ERR_X509_INVALID_SIGNATURE + MBEDTLS_ERR_ASN1_INVALID_DATA
+
+## This edge case currently fails to parse due to an ASN.1 bug.
+## It doesn't matter as far as X.509 is concerned since the signature would
+## not be valid anyway.
+## https://github.com/ARMmbed/mbed-crypto/pull/75 will fix this bug and make
+## this test case pass.
```
Comment (Patater) [inline tests/suites/test_suite_x509parse.data]: So, we should merge this PR at the same time as the Crypto one to minimize how long the test is breaking on this one X.509 test, right?

Comment (gilles-peskine-arm) [reply tests/suites/test_suite_x509parse.data]: No: I left the test commented out. First we merge this, then we merge the crypto PR, then we uncomment this test.

Comment (Patater) [reply tests/suites/test_suite_x509parse.data]: OK. In my first round of review, I noticed this was commented out. On my laptop, the line wrapped in a way that made this less obvious.

Screenshot 2019-09-30 at 16 19 32

**Pull request merged**